File: concrete/src/test/java/pcb/uwu/amount/base/CandelasTest.java
package pcb.uwu.amount.base;

import org.junit.Test;
import pcb.uwu.amount.derived.fundamental.area.SquareMeters;
import pcb.uwu.amount.derived.optics.Nits;
import pcb.uwu.amount.derived.scalar.Steradians;

import static java.math.MathContext.DECIMAL64;
import static org.junit.Assert.assertEquals;

public class CandelasTest {
	private static final Candelas candelas = new Candelas(18);

	@Test
	public void multipliedBySolidAngle() {
		assertEquals("54 lm", candelas.multipliedBy(new Steradians(3), DECIMAL64).toString());
	}

	@Test
	public void dividedByArea() {
		assertEquals("6 nit", candelas.dividedBy(new SquareMeters(3), DECIMAL64).toString());
	}

	@Test
	public void dividedByLuminance() {
		assertEquals("3 m²", candelas.dividedBy(new Nits(6), DECIMAL64).toString());
	}
}

File: concrete/src/test/java/pcb/uwu/amount/base/CandelasTest.kt
package pcb.uwu.amount.base

import org.junit.Assert.assertEquals
import org.junit.Test
import pcb.uwu.amount.derived.fundamental.area.SquareMeters
import pcb.uwu.amount.derived.optics.Nits
import pcb.uwu.amount.derived.scalar.Steradians

class CandelasTest
{
    @Test
    fun multipliedBySolidAngle() =
        assertEquals("54 lm",
                     (Candelas(18) * Steradians(3)).toString())

    @Test
    fun dividedByArea() =
        assertEquals("6 nit",
                     (Candelas(18) / SquareMeters(3)).toString())

    @Test
    fun dividedByLuminance() =
        assertEquals("3 m²",
                     (Candelas(18) / Nits(6)).toString())
}